GammaGeeks/tura-backend
# Verify email endpoint

**What does this PR do?**

This PR implements the email verification functionality for new users of the Barefoot Nomad application, allowing them to confirm their email address after signing up. This process helps secure the account and ensures users can access the application's full features.

**Description of the task to be completed:**

- Send an email with a verification link to the user’s registered email upon successful signup.
- Create a verification endpoint that the link directs to, confirming the user’s email address.
- Mark the user's email as verified in the system once the link is successfully accessed.
- Display an error message if the link is expired or invalid, with an option to request a new verification email.
- Restrict access to certain features until the email is verified.

**How should this be manually tested?**

1. Clone the repository and navigate to the project directory.
2. Start the backend server and ensure dependencies are installed.
3. Sign up with a new user account, and check the registered email for a verification link.
4. Click the verification link to confirm email verification and check that the account status updates to verified.
5. Attempt to access the verification link again to confirm that an expired or used link displays an appropriate error message.
6. Use the option to request a new verification email to test the re-sending functionality.

**What are the relevant pivotal tracker stories?**

- **Title**: Verify Email after SignUp
- **ID**: [#188572711](https://www.pivotaltracker.com/n/projects/2725084/stories/188572711)
